akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ClusterShardingImpl.scala

```diff
@@ -53,6 +55,7 @@ import akka.util.JavaDurationConverters._
     extends ShardingMessageExtractor[Any, M] {
   override def entityId(message: Any): String = {
     message match {
+      case EntityEnvelope(entityId, _)   => entityId
```

Comment:
we will be able to remove this if we manage to do the full conversion and remove the serialization.